lab_2/main.py

```diff
@@ -27,32 +27,99 @@ def read_ecg_rr_file(file_path: Path):
         rr_signal.append(numeric_value)
     return rr_signal
 
+# def test_input_of_float(input:list):
+#     if type(input) != list or len(input)==0:
+#         return None
+#     for i in input:
+#         if type(i) != float:
+#             return None
+
+def calculating_rrd (signal: list):
+    rrd=[]
+    for i in range(0,len(signal [1:])):
+        rrd_i=signal[i]-signal[i+1]
+        rrd.append(rrd_i)
+    return rrd
 
 # Lab 2 implementation goes below
-
 def calculate_sdnn(signal: list):
     """Calculating SDNN for RR-intervals from the list"""
-    pass
-
+    if type(signal) != list:
+        return None
+    if len(signal) == 0:
+        return None
+    for i in signal:
+        if type(i) != float:
+            return None
+    mean_rr=sum(signal)/len(signal)
+    sum_diff=0
+    for i in signal:
+        rri_rrmean=(i-mean_rr)**2
+        sum_diff+=rri_rrmean
+    return round((sum_diff/len(signal))**0.5, 2)
 
 def calculate_rmssd(signal: list):
     """Calculating RMSSD for RR-intervals from the list"""
-    pass
-
+    if type(signal) != list:
+        return None
+    if len(signal) == 0:
+        return None
+    for i in signal:
+        if type(i) != float:
+            return None
+    sum = 0
+    rrd = calculating_rrd(signal)
+    for i in rrd:
+        skobki = i**2
+        sum+=skobki
+    return round((sum/(len(signal)-1))**0.5, 2)
 
 def calculate_sdsd(signal: list):
     """Calculating SDSD for RR-intervals from the list"""
-    pass
-
+    if type(signal) != list:
+        return None
+    if len(signal) == 0:
+        return None
+    for i in signal:
+        if type(i) != float:
+            return None
+    rrd=calculating_rrd(signal)
+    mean_rrd=sum(rrd)/len(rrd)
+    brackets_sum=0
+    for i in rrd:
+        brackets_sum+=(i-mean_rrd)**2
+    return round((brackets_sum/(len(rrd[1:])))**0.5, 2)
 
 def calculate_nn_pnn(signal: list, threshold: int):
     """Calculating NN and pNN for RR-intervals from the list"""
-    pass
-
+    if type(signal) != list:
+        return None
+    if len(signal) == 0:
+        return None
+    for i in signal:
+        if type(i) != float:
+            return None
+    if type(threshold) != int:
+        return None
+    rrd_list=calculating_rrd(signal)
+    result=[]
+    nn=[]
+    for i in rrd_list:
+        if abs(i) <= threshold:
+            nn.append(i)
+    result.append(len(nn))
+    result.append(round(len(nn)/len(rrd_list), 2))
+    return result
```

Comment:
if we return several values with a comma, what will be the type? Is it a list??

```suggestion
    return len(nn), round(len(nn)/len(rrd_list), 2)
```